aic-emu/GfxHandler.cc

```diff
@@ -0,0 +1,542 @@
+    GFX_LOG_STATUS((m_device_fd < 0), "open device");
+    if (!GFX_SUCCESS(m_device_fd < 0))
+        goto exit_logic;
+
+    //Get version info
+    struct drm_version drm_ver;
+    memset(&drm_ver, 0, sizeof(drm_ver));
+
+    sts = RunIoctl(DRM_IOCTL_VERSION, &drm_ver);
+
+    GFX_LOG_FAILURE(sts != GFX_OK, "drmIoctl DRM_IOCTL_VERSION first");
+    if (!GFX_SUCCESS(sts))
+        goto exit_logic;
+
+    //Allocate memory and call the ioctl again to actually get the name
+    if (drm_ver.name_len > 0)
+    {
+        drm_ver.name = (char *)malloc(drm_ver.name_len + 1);
+
+        sts = RunIoctl(DRM_IOCTL_VERSION, &drm_ver);
+
+        GFX_LOG_FAILURE(sts != GFX_OK, "drmIoctl DRM_IOCTL_VERSION second");
+
+        if (drm_ver.name_len >=0)
+            drm_ver.name[drm_ver.name_len] = '\0';
+
+        m_device_props.version = (const char *) drm_ver.name;
+
+        if (drm_ver.name)
+            free(drm_ver.name);
+
+        if (!GFX_SUCCESS(sts))
+            goto exit_logic;
+    }
+
+    //Get Chipset ID
+    sts = GetDrmParam(I915_PARAM_CHIPSET_ID, &m_device_props.chipsetId);
+    GFX_LOG_FAILURE(sts != 0, "drmIoctl I915_PARAM_CHIPSET_ID");
+    if (!GFX_SUCCESS(sts))
+        goto exit_logic;
+
+
+    //MMAP offset
+    sts = GetDrmParam(I915_PARAM_MMAP_GTT_VERSION, &mmap_gtt_version);
+    if (!GFX_SUCCESS(sts))
+        goto exit_logic;
+    m_device_props.hasMMAPoffset = (mmap_gtt_version >= 4);
+
+    if (!m_device_props.hasMMAPoffset)
+    {
+        std::cout << "Error: Device doesn't have MMAPoffset function." << std::endl;
+        sts = GFX_FAIL;
+    }
+
+exit_logic:
+    if (!GFX_SUCCESS(sts))
+    {
+        if (m_device_fd >= 0)
+        {
+            close(m_device_fd);
+            m_device_fd = -1;
+        }
+    }
+
+    std::cout << std::string(__FUNCTION__) << ": drmVersion.name = " << m_device_props.version << "\n"
+              << "\tchipsetID = 0x" << std::hex  << m_device_props.chipsetId << std::dec << "\n"
+              << "\thasMMAPoffset = " << m_device_props.hasMMAPoffset << std::endl;
+
+    return (GfxStatus)sts;
+}
+
+GfxStatus GfxHandler::GfxClose()
+{
+    GfxStatus sts = GFX_OK;
+
+    if (m_device_fd >= 0)
+    {
+        close(m_device_fd);
+        m_device_fd = -1;
+    }
+    return sts;
+}
+
+GfxStatus GfxHandler::DetermineSurfaceParams(SurfaceParams_t& surf, unsigned width, unsigned height, unsigned format)
+{
+    if (width == 0 || height == 0 || format == 0)
+    {
+        std::cout << "Error: Unexpected surface parameter: width = " << width
+                  << ", height = " << height << ", format = " << format << std::endl;
+
+        return GFX_FAIL;
+    }
+
+	uint32_t horizontal_alignment = 4;
+	uint32_t vertical_alignment = 4;
+
+    memset(&surf, 0, sizeof(SurfaceParams_t));
+
+    surf.width  = width;
+    surf.height = height;
+    surf.format = format;
+    surf.pitch  = surf.width;
+
+    surf.tilingFormat = GetTilingFormat(format);
+
+	switch (surf.tilingFormat)
```

Comment:
fix alignment. You have tabs here and below

Reply:
done